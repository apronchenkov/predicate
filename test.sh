#!/bin/sh

# Comparison
## ==
./predicate "x == y" \
    "x = 0, y = 0" \
    "x = 0, y = 1" \
    "x = 1, y = 0" \
    "x = 1, y = 1"

## !=
./predicate "x == y" \
    "x = 0, y = 0" \
    "x = 0, y = 1" \
    "x = 1, y = 0" \
    "x = 1, y = 1"

## in
./predicate "x in {a, b}" \
    "x = a" \
    "x = b" \
    "x = c"

## in
./predicate "x not in {a, b}" \
    "x = a" \
    "x = b" \
    "x = c"

## x
./predicate "x" \
    "x = ''" \
    "x = 0" \
    "x = 1"


# (x)
./predicate "(x)" \
    "x = ''" \
    "x = 0" \
    "x = 1"


# not
./predicate "not x" \
    "x = ''" \
    "x = 0" \
    "x = 1"


# and
./predicate "x and y" \
    "x = '', y = ''" \
    "x = '', y = 1" \
    "x = 1, y = ''" \
    "x = 1, y = 1"


# or
./predicate "x or y" \
    "x = '', y = ''" \
    "x = '', y = 1" \
    "x = 1, y = ''" \
    "x = 1, y = 1"


# A free form
./predicate "kind in {'street', 'district'} and 'TR' == country or fallback" \
    "kind=locality, country=RU" \
    "kind=locality, country=TR" \
    "kind=district, country=RU" \
    "kind=district, country=TR" \
    "kind=street, country=RU" \
    "kind=street, country=TR" \
    "kind=locality, country=TR, fallback=1" \
    "kind=district, country=RU, fallback=1" \
    "kind=district, country=TR, fallback=1" \
    "kind=street, country=RU, fallback=1" \
    "kind=street, country=TR, fallback=1" \
    "fallback=1"
