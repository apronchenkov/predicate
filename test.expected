x == y: x = 0, y = 0: True
x == y: x = 0, y = 1: False
x == y: x = 1, y = 0: False
x == y: x = 1, y = 1: True
x == y: x = 0, y = 0: True
x == y: x = 0, y = 1: False
x == y: x = 1, y = 0: False
x == y: x = 1, y = 1: True
x in {a, b}: x = a: True
x in {a, b}: x = b: True
x in {a, b}: x = c: False
x not in {a, b}: x = a: False
x not in {a, b}: x = b: False
x not in {a, b}: x = c: True
x: x = '': False
x: x = 0: True
x: x = 1: True
(x): x = '': False
(x): x = 0: True
(x): x = 1: True
not x: x = '': True
not x: x = 0: False
not x: x = 1: False
x and y: x = '', y = '': False
x and y: x = '', y = 1: False
x and y: x = 1, y = '': False
x and y: x = 1, y = 1: True
x or y: x = '', y = '': False
x or y: x = '', y = 1: True
x or y: x = 1, y = '': True
x or y: x = 1, y = 1: True
kind in {'street', 'district'} and 'TR' == country or fallback: kind=locality, country=RU: False
kind in {'street', 'district'} and 'TR' == country or fallback: kind=locality, country=TR: False
kind in {'street', 'district'} and 'TR' == country or fallback: kind=district, country=RU: False
kind in {'street', 'district'} and 'TR' == country or fallback: kind=district, country=TR: True
kind in {'street', 'district'} and 'TR' == country or fallback: kind=street, country=RU: False
kind in {'street', 'district'} and 'TR' == country or fallback: kind=street, country=TR: True
kind in {'street', 'district'} and 'TR' == country or fallback: kind=locality, country=TR, fallback=1: True
kind in {'street', 'district'} and 'TR' == country or fallback: kind=district, country=RU, fallback=1: True
kind in {'street', 'district'} and 'TR' == country or fallback: kind=district, country=TR, fallback=1: True
kind in {'street', 'district'} and 'TR' == country or fallback: kind=street, country=RU, fallback=1: True
kind in {'street', 'district'} and 'TR' == country or fallback: kind=street, country=TR, fallback=1: True
kind in {'street', 'district'} and 'TR' == country or fallback: fallback=1: True
